contracts/BosonRouter.sol

```diff
@@ -1247,23 +1232,31 @@ contract BosonRouter is
     /**
      * @notice finalizes creating of conditional order
      * @param _tokenIdSupply    ID of the supply token
-     * @param _gateAddress address of a gate contract that will handle the interaction between the BosonRouter contract and the conditional token,
+     * @param _conditionalCommitInfo struct that contains data pertaining to conditional commit:
+     *
+     * uint256 conditionalTokenId - Id of the conditional token, ownership of which is a condition for committing to redeem a voucher
+     * in the voucher set created by this function.
+     *
+     * uint256 threshold - the number that the balance of a tokenId must be greater than or equal to. Not used for OWNERSHIP condition
+     *
+     * Condition condition - condition that will be checked when a user commits using a conditional token
+     *
+     * address gateAddress - address of a gate contract that will handle the interaction between the BosonRouter contract and the conditional token,
      * ownership of which is a condition for committing to redeem a voucher in the voucher set created by this function.
-     * @param _conditionalTokenId Id of the conditional token, ownership of which is a condition for committing to redeem a voucher
-     * @param _condition condition that will be checked when a user commits using a conditional token
-     * @param _registerConditionalCommit indicates whether Gate.registerVoucherSetId should be called. Gate.registerVoucherSetId can also be called separately
+     *
+     * bool registerConditionalCommit - indicates whether Gate.registerVoucherSetId should be called. Gate.registerVoucherSetId can also be called separately
      */
-    function finalizeConditionalOrder(uint256 _tokenIdSupply, address _gateAddress, uint256 _conditionalTokenId, Condition _condition, uint256 _threshold, bool _registerConditionalCommit) internal {
-        voucherSetToGateContract[_tokenIdSupply] = _gateAddress;
+    function finalizeConditionalOrder(uint256 _tokenIdSupply, ConditionalCommitInfo calldata _conditionalCommitInfo) internal {
+        voucherSetToGateContract[_tokenIdSupply] = _conditionalCommitInfo.gateAddress;
 
-        emit LogConditionalOrderCreated(_tokenIdSupply, _gateAddress, _conditionalTokenId, _condition, _threshold);
+        emit LogConditionalOrderCreated(_tokenIdSupply, _conditionalCommitInfo.gateAddress, _conditionalCommitInfo.conditionalTokenId, _conditionalCommitInfo.condition, _conditionalCommitInfo.threshold);
 
-        if (_registerConditionalCommit) {
-            IGate(_gateAddress).registerVoucherSetId(
+        if (_conditionalCommitInfo.registerConditionalCommit) {
+            IGate(_conditionalCommitInfo.gateAddress).registerVoucherSetId(
```

Comment:
Is there any reason not to just change the `registerVoucherSetId` method to take a `ConditionalCommitInfo` argument rather than passing its first three arguments individually and then creating it with literal defaults for the last two args?

Reply:
No. No real reason. I'll change it.